### PR TITLE
add initial docker build, scripting and travis stage

### PR DESCRIPTION
**Pull request by sbaier1**
**Motivation**

Make the HiveMQ CE more accessible by offering an official HiveMQ CE docker image.

**Changes**
add scripting and dockerfile

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -61,6 +61,33 @@ If you want to connect devices on external networks to HiveMQ CE, please make su
 
 Just in time builds for current branches on this repository and for specific commits are available https://hivemq.github.io/nightly-builds/[here].
 
+== Run with Docker
+
+The current state of the `master` branch will be pushed to the https://hub.docker.com/r/hivemq/hivemq-ce[hivemq/hivemq-ce] repository.
```
Comment (schaebo) [inline README.adoc]: The current state of the `master` branch will be pushed to the
->
All releases as well as the current state of the `master` branch are available in the https://hub.docker.com/r/hivemq/hivemq-ce[hivemq/hivemq-ce] repository on dockerhub.

File: README.adoc [hunk 2]
```diff
@@ -61,6 +61,33 @@ If you want to connect devices on external networks to HiveMQ CE, please make su
...
+To execute this image, simply run the following command:
+
+[source,bash]
+----
+docker run --name hivemq-ce -e HIVEMQ_LOG_LEVEL=INFO -d -p 1883:1883 /hivemq/hivemq-ce
```
Comment (schaebo) [inline README.adoc]: The "-e HIVEMQ_LOG_LEVEL=INFO" is the default anyway. So i think we can skip it in the default command.

File: .travis.yml [hunk 3]
```diff
@@ -9,13 +9,18 @@ jdk:
...
+    if: branch = master && type != pull_request
 
 jobs:
   include:
     - stage: test
       script: ./gradlew test
     - stage: build
       script: ./gradlew packaging
+    - stage: docker
+      install: true
```
Comment (schaebo) [inline .travis.yml]: is this the same as `install: skip` ?

File: docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,69 @@
+FROM busybox AS unzip
```
Comment (schaebo) [inline docker/Dockerfile]: A comment would be nice, why we have an additional stage here.

File: .travis.yml (hunk 3 above)
Comment (SgtSilvio) [reply .travis.yml]: yes, it is the same
I think skip is newer and more descriptive

Comment (SgtSilvio) [reply .travis.yml]: We should add `install: skip` to all stages

File: README.adoc (hunk 2 above)
Comment (sbaier1) [reply README.adoc]: i would've left it there simply to demonstrate how to manipulate the log level.

File: .travis.yml (hunk 3 above)
Comment (sbaier1) [reply .travis.yml]: added it to all stages

File: README.adoc (hunk 2 above)
Comment (sbaier1) [reply README.adoc]: @schaebo what do you think?

File: README.adoc [hunk 5]
```diff
@@ -61,6 +61,33 @@ If you want to connect devices on external networks to HiveMQ CE, please make su
...
+
+All releases as well as the current state of the  `master` are available in the https://hub.docker.com/r/hivemq/hivemq-ce[hivemq/hivemq-ce] repository on DockerHub.
+To execute this image, simply run the following command:
+
+[source,bash]
+----
+docker run --name hivemq-ce -e HIVEMQ_LOG_LEVEL=INFO -d -p 1883:1883 /hivemq/hivemq-ce
+----
+
+=== Build Docker image manually
```
Comment (schaebo) [inline README.adoc]: This part should be below "Building From Source"."Start" imho

File: README.adoc (hunk 2 above)
Comment (schaebo) [reply README.adoc]: Let's try to keep the command as simple as possible.

Maybe add a line in the "Build Docker image manually" like.
To change the default log level you can set the environment variable `HIVEMQ_LOG_LEVEL` on the container.

File: .travis.yml [hunk 6]
```diff
@@ -9,13 +9,20 @@ jdk:
 stages:
   - test
   - build
+  - name: docker
+    if: branch = master && type != pull_request
```
Comment (SgtSilvio) [inline .travis.yml]: I don't think this runs for releases, which are tag builds and do not have a branch

**Comment by schettada** [comment]
Any idea when this will be merged to Release and a docker image available?

**Pull request merged**